src/cpp/core/file_lock/LinkBasedFileLock.cpp

```diff
@@ -277,19 +277,49 @@ Error writeLockFile(const FilePath& lockFilePath)
             proxyPath.absolutePathNative().c_str(),
             lockFilePath.absolutePathNative().c_str());
    
-   // log errors (remove this if it is too noisy on NFS)
+   // detect link failure
    if (status == -1)
    {
-      int errorNumber = errno;
-      
       // verbose logging
+      int errorNumber = errno;
       LOG("ERROR: ::link() failed (errno " << errorNumber << ")" << std::endl <<
           "Attempted to link:" << std::endl << " - " <<
           "'" << proxyPath.absolutePathNative() << "'" <<
           " => " <<
           "'" << lockFilePath.absolutePathNative() << "'");
       
-      LOG_ERROR(systemError(errorNumber, ERROR_LOCATION));
+      // if this failed, we should still make a best-effort attempt to acquire
+      // a lock by creating a file using O_CREAT | O_EXCL. note that we prefer
+      // ::link() since older NFSes provide more guarantees as to its atomicity,
+      // but not all NFS support ::link()
```

Comment:
Do we know the actual `errno` returned when NFS doesn't support `link`? 

If we do I think we should scope this retry so that it only activates in this specific case. If we can't do this we could just blacklist the errnos which we could be returned by link() that preclude the possibility of a retry working (permission denied, etc.)